contracts/utils/convex/ConvexStakingWrapperYield.sol

```diff
@@ -0,0 +1,119 @@
+//https://etherscan.io/address/0x3ba207c25a278524e1cc7faaea950753049072a4#code
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+pragma experimental ABIEncoderV2;
+
+import '@yield-protocol/vault-interfaces/ICauldron.sol';
+import '@yield-protocol/vault-interfaces/DataTypes.sol';
+import './ConvexStakingWrapper.sol';
+
+/// @title Convex staking wrapper for Yield platform
+/// @notice Enables use of convex LP positions as collateral while still receiving rewards
+contract ConvexStakingWrapperYield is ConvexStakingWrapper {
+    using TransferHelper for IERC20;
+
+    /// @notice Mapping to keep track of the user & their vaults
+    mapping(address => bytes12[]) public vaults;
+
+    ICauldron cauldron;
+
+    /// @notice Event called when a vault is set for a user
+    /// @param account The account for which vault is set
+    /// @param vault The vaultId
+    event VaultSet(address account, bytes12 vault);
+
+    constructor(
+        address curveToken_,
+        address convexToken_,
+        address convexPool_,
+        uint256 poolId_,
+        address join_,
+        ICauldron cauldron_,
+        address timelock_
+    ) {
+        owner = address(timelock_);
+        emit OwnershipTransferred(address(0), owner);
+        _tokenname = string(abi.encodePacked('Staked ', ERC20(convexToken_).name(), ' Yield'));
+        _tokensymbol = string(abi.encodePacked('stk', ERC20(convexToken_).symbol(), '-yield'));
+        isShutdown = false;
+        isInit = true;
+        curveToken = curveToken_;
+        convexToken = convexToken_;
+        convexPool = convexPool_;
+        convexPoolId = poolId_;
+        collateralVault = join_; //TODO: Add the join address
+        cauldron = cauldron_;
+
+        //add rewards
+        addRewards();
+        setApprovals();
+    }
+
+    function setCollateralVault(address join_) external {
```

Comment:
Please make this `auth` instead, and emit an event at the end.

Please also rename this function as `point`. We use that as the function name every time time that we allow to change the orchestration after deployment.